lib/reversetunnel/agentpool.go

```diff
@@ -632,14 +631,25 @@ func (c *agentPoolRuntimeConfig) restrictConnectionCount() bool {
 	return c.tunnelStrategyType == types.ProxyPeering
 }
 
-// useReverseTunnelV2 returns true if reverse tunnel should be used.
-func (c *agentPoolRuntimeConfig) useReverseTunnelV2() bool {
+// alpnDialerConfig creates a config for ALPN dialer.
+func (c *agentPoolRuntimeConfig) alpnDialerConfig(getClusterCAs client.GetClusterCAsFunc) client.ALPNDialerConfig {
 	c.mu.RLock()
 	defer c.mu.RUnlock()
-	if c.isRemoteCluster {
-		return false
+
+	protocols := []alpncommon.Protocol{alpncommon.ProtocolReverseTunnel}
+	if !c.isRemoteCluster && c.tunnelStrategyType == types.ProxyPeering {
```

Comment:
can we use old perserve old function without a local: 

```suggestion
	if v.useReverseTunnel2() {
```